### ISSUE TITLE
Uncaught TypeError: Super expression must either be null or a function, not undefined on EnhancerProvider

### ISSUE THREAD

**Issue by ZSeba** [effektif/react-mentions#169]
Hello, I'm getting this error whenever I try to render my app.

` Uncaught TypeError: Super expression must either be null or a function, not undefined` with EnhancerProvider.js

My development environment:

- React 15.5.4
- Webpack 2.6.1
- TypeScript 2.3.4

My component looks like this: 
```
<form className="form-inline" onSubmit={this.handleSubmit}>
        <div className = "form-group">
          <MentionsInput className="form-control" value={this.state.messageValue} onChange={this.handleChange}>
              <Mention trigger="@" data={this.state.availableUsers}/>
          </MentionsInput>
          <button className="btn btn-primary" type="submit">
            Send
          </button>
        </div>
      </form>
```

Webpack works fine by the way, this only happens when trying to display the component on my browser
Also,  I tried using a className in MentionsInput to use plain old CSS, but it's not working either.

Thanks ! and apologies in advance if  this turns out to be some dumb mistake :sweat_smile:

**Comment by frontendphil** [comment]
Haven't seen this one before. @jfschwarz sounds like something with `substyle`. @ZSeba do you use the `<EnhancerProvider />` from substyle?

**Comment by jfschwarz** [comment]
Yeah, sounds like you're trying to use substyle's `<EnhancerProvider>` in an incorrect way. Your code snippet does not include this part, though. If you want to use plain old css, just remove the `<EnhancerProvider>` altogether.

**Comment by ZSeba** [comment]
But I'm not using `<EnhancerProvider>` at all, just this form component I showed above

**Comment by jfschwarz** [comment]
In that case I'm confused about your bug report above:

> Uncaught TypeError: Super expression must either be null or a function, not undefined with EnhancerProvider.js

What is `EnhancerProvider.js` and why is it executed if you do not render an `<EnhancerProvider>` element? (react-mentions does never render this element.)

**Comment by ZSeba** [comment]
Yeah, I'm confused too. I'm not using `<EnhancerProvider>` anywhere in my code but that's the error I get ("frontapp" being my webpack output )

![eperror](https://user-images.githubusercontent.com/567132/27872341-7cc37688-6176-11e7-9843-2f7600ad728c.png)

And `react-mentions` does have `substyle` as a dependency, from which EnhancerProvider.js comes.

**Comment by jfschwarz** [comment]
Ok, so if you do not render an `<EnhancerProvider>` it seems to be sufficient that EnhancerProvider.js is merely required to produce the error in your environment. `EnhancerProvider` extends `React.PureComponent`, which has been added with version 15.3. Since the error message indicates that the super class is `undefined` please check if your version of React really exports the `PureComponent` class.

**Comment by jfschwarz** [comment]
Closing this for inactivity

**Comment by Mrjaco12** [comment]
I've just run into this issue and the fix for me was to update react to >= 15.3. I'm not too familiar with js package management so I can't say what the best way to handle this is but it seems like since some of the dependencies for this project require react >=15.3 this project should as well.

**Comment by ZSeba** [comment]
Thanks for your suggestions. I ended up using draftjs instead 😞
